js/react-tweek/src/createUseTweekValue.ts

```diff
@@ -60,7 +62,12 @@ export const createUseTweekValue = (
           }
         })
       );
-    }, [tweekRepository, keyPath, defaultValue]);
+    }, args);
+
+    if (!isEqual(args, storedArgs.current)) {
+      storedArgs.current = args;
+      return getTweekValue();
```

Comment:
maybe use the same logic of the reducer, if the current value is equal to the new value return the old value for reference equality